storage/tianmu/core/item_tianmu_field.cpp

```diff
@@ -66,10 +66,15 @@ void Item_tianmufield::SetType(DataType t) {
   tianmu_type = t;
   switch (tianmu_type.valtype) {
     case DataType::ValueType::VT_FIXED:
-      if (tianmu_type.IsInt())
-        ivalue = new Item_int(static_cast<longlong>(0));
-      else
+      if (tianmu_type.IsInt()) {
+        if (ifield->unsigned_flag) {
```

Comment:
Pls, refine these `if` statements. Make it more simple and concise. Thxs.

Reply:
value type is :`VT_FIXED` , but in Item_float, we use `not_fixed_dec`.
```
#define NOT_FIXED_DEC 31

/*
  The longest string my_fcvt can return is 311 + "precision" bytes.
  Here we assume that we never cal my_fcvt() with precision >= NOT_FIXED_DEC
  (+ 1 byte for the terminating '\0').
*/
#define FLOATING_POINT_BUFFER (311 + NOT_FIXED_DEC)

/*
  We want to use the 'e' format in some cases even if we have enough space
  for the 'f' one just to mimic sprintf("%.15g") behavior for large integers,
  and to improve it for numbers < 10^(-4).
  That is, for |x| < 1 we require |x| >= 10^(-15), and for |x| > 1 we require
  it to be integer and be <= 10^DBL_DIG for the 'f' format to be used.
  We don't lose precision, but make cases like "1e200" or "0.00001" look nicer.
*/
#define MAX_DECPT_FOR_F_FORMAT DBL_DIG

/*
  The maximum possible field width for my_gcvt() conversion.
  (DBL_DIG + 2) significant digits + sign + "." + ("e-NNN" or
  MAX_DECPT_FOR_F_FORMAT zeros for cases when |x|<1 and the 'f' format is used).
*/
#define MY_GCVT_MAX_FIELD_WIDTH (DBL_DIG + 4 + MY_MAX(5, MAX_DECPT_FOR_F_FORMAT)) \

```

Reply:
got it. 
Solving problems in a better way.